CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+```
+
+### Make the necessary changes.
+
+### Stage your changes and commit.
+
+```bash
+git add . # Stages all the changes
+git commit -m "<your_commit_message>"
+```
+
+Your commit message should be something which gives concise idea of the issue you are solving.
+
+We implement the Conventional Commits specification for commit messages. This specification is a lightweight convention on top of commit messages. It provides an easy set of rules for creating an explicit commit history; which makes it easier to write automated tools on top of. This convention dovetails with SemVer, by describing the features, fixes, and breaking changes made in commit messages.
+
+The commit message should be structured as follows:
+
+```
+<type>(optional scope): <description>
+```
+
+Push your local commits to the remote repository.
+
+```bash
+git push origin your-branch-name
+```
+
+**8.** Create a new [pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) from `your-branch-name`
+
+🎉 Congratulations! You've made your first pull request! Now, you should just wait until the maintainers review your pull request.
+
+## ✨ Understanding the project
+
+## Prerequites
+
+This project uses [pnpm](https://pnpm.io/) as its package manager. If you had [nodejs](https://nodejs.org/en) installed, this is how you can install [pnpm](https://pnpm.io/) globally:
+
+```
+npm i -g pnpm # Install pnpm
+```
+
+The project uses a [monorepo](https://monorepo.tools/) structure. It is divided into 3 main folders:
+
+- `packages` - This folder contains all the packages used within the Million.js library
+- `test` - This folder contains all the tests written for the Million.js library
+- `website` - This folder contains the code for the website [Million.js](https://millionjs.org/)
+
+### `packages` folder
+
+The `packages` folder contains all the code related to the Million.js package. The package is built  using Typescript.
+
+The source code is divided into five main folders:
+
+* `compiler` - This folder contains all the code related to the Million.js React and Next.js compilers
+* `jsx-runtime` - This folder contains the code related to the runtime code for jsx.
+* `million` - This folder contains the code for the core Million.js package. The implementation of the optimized array rendering (`<For/>`), `block()` and virtual DOM for React are in here.
+* `next` - This folder contains the Million.js support package for Next.js
+* `react` - This folder contains the Million.js support package for React
+
+To run the project locally, you need to install the dependencies.
+
+```bash
+pnpm install # Install dependencies
+cd {whichever of the packages}
+pnpm {command} # Run the project
+```
+
+`test` folder
+
+This folder contains the code for the tests for the implemetation of core Million.js features like the "map-array" with ``<For/>, ``block() and so on. The test use the [Vitest Framework](https://vitest.dev/).
```

Comment:
Issue with misaligned codeblocks, instead of ``<For/>, do `<For />`

Reply:
Done